homeassistant/components/vorwerk/config_flow.py

```diff
@@ -0,0 +1,136 @@
+"""Config flow to configure Vorwerk integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import pybotvac
+from pybotvac.exceptions import NeatoException
+from requests.models import HTTPError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_CODE, CONF_EMAIL, CONF_TOKEN
+
+from homeassistant.data_entry_flow import FlowResult
+
+# pylint: disable=unused-import
+from .const import (
+    VORWERK_CLIENT_ID,
+    VORWERK_DOMAIN,
+    VORWERK_ROBOT_ENDPOINT,
+    VORWERK_ROBOT_NAME,
+    VORWERK_ROBOT_SECRET,
+    VORWERK_ROBOT_SERIAL,
+    VORWERK_ROBOT_TRAITS,
+    VORWERK_ROBOTS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class VorwerkConfigFlow(config_entries.ConfigFlow, domain=VORWERK_DOMAIN):
+    """Vorwerk integration config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize the config flow."""
+        self._email: str | None = None
+        self._session = VorwerkSession()
+
+    async def async_step_user(self, user_input=None):
+        """Step when user initializes a integration."""
+
+        if user_input is not None:
+            self._email = user_input.get(CONF_EMAIL)
+            if self._email:
+                await self.async_set_unique_id(self._email)
+                self._abort_if_unique_id_configured()
+                return await self.async_step_code()
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_EMAIL): str,
+                }
+            ),
+        )
+
+    async def async_step_code(self, user_input: dict[str, Any] = None) -> FlowResult:
+        """Step when user enters OTP Code from email."""
+        assert self._email is not None  # typing
+        errors = {}
+        code = user_input.get(CONF_CODE) if user_input else None
+        if code:
+            try:
+                robots = await self.hass.async_add_executor_job(
+                    self._get_robots, self._email, code
+                )
+                return self.async_create_entry(
+                    title=self._email,
+                    data={
+                        CONF_EMAIL: self._email,
+                        CONF_TOKEN: self._session.token,
+                        VORWERK_ROBOTS: robots,
+                    },
+                )
```

Comment:
Move outside of try

Reply:
Then an extra `if` is needed. I don't think that makes the code more readable.

Reply:
It is compulsory to move it out of the try block, but you can use `try...except...else` form

Reply:
@epenet I will try that.